components/zip321/CHANGELOG.md

```diff
@@ -8,10 +8,28 @@ indicated by the `PLANNED` status in order to make it possible to correctly
 represent the transitive `semver` implications of changes within the enclosing
 workspace.
 
-## [Unreleased]
+## [0.7.0] - PENDING
 
 ### Changed
 - MSRV is now 1.85.1.
+- `zip321::Payment::amount` now returns `Option<Zatoshis>` instead of `Zatoshis`.
+  Previously, the amount field of the payment would be set to zero if no amount
+  parameter was present; however, this behavior is not specified by ZIP 321.
+  The semantics of this field have now been altered such that if no amount is
+  specified by a payment request, this should be interpreted to mean that the
+  sender of the transaction should specify an amount for the payment.
+- `zip321::Payment::new` now takes its `amount` parameter as `Option<Zatoshis>`
+  instead of `Zatoshis`. It now returns `None` if the payment requests that a
+  zero-valued output OR a memo be sent to a transparent-only address.
+- `zip321::Zip321Error` has added variant `Zip321Error::ZeroValuedTransparentOutput`.
+  Zero-valued transparent outputs are rejected by the Zcash consensus rules,
```

Comment:
There is no such consensus rule. Zcash is in this respect like Bitcoin, which also allows zero-valued UTXOs: https://bitcoin.stackexchange.com/questions/104937/transaction-outputs-with-value-0

There is a standard rule in both Zcash and Bitcoin that transactions with "dust" outputs (for Zcash, call this "transparent dust" to avoid ambiguity), should not be mined. So the restriction on constructing zero-valued transparent outputs is perfectly reasonable, but it should be described accurately as a consequence of a standard rule, not a consensus rule.

```suggestion
  Zero-valued transparent outputs are rejected by the Zcash standard rules,
```